src/components/NewItemForm.jsx

```diff
@@ -36,14 +59,39 @@ const NewItemForm = () => {
   };
 
   const handleDialogClose = (dialogState) => {
-    setAnchorEl(null);
+    handleMenuClose();
+    clearAllErrors();
+    clearAllInputs();
     dialogState(false);
   };
 
+  const navigate = useNavigate();
   // Collections Form
   const [collectionName, setCollectionName] = useState("");
-  const onSaveCollection = () => {
-    console.log(collectionName);
+  const [collectionError, setCollectionError] = useState(clearError);
+  const onSaveCollection = async () => {
+    if (collectionName === "") {
+      setCollectionError(requiredFieldsError);
+      return;
+    } else {
+      setCollectionError(clearError);
+    }
+
+    try {
+      let response = await postCollections(collectionName);
+      setCollectionName("");
+      handleDialogClose(setCollectionsDialogOpen);
+      (prevData) => [...prevData, data1];
+      setCollectionArray((currentCollections) => [
+        ...currentCollections,
+        response.data,
+      ]);
+      // Get new vault collection items
+      getVaultCollections();
+    } catch (error) {
+      networkErrorHandler(error);
```

Comment:
Nice I was just going to ask if there was a way to abstract the try catch somehow to handle the errors since we will be needing the error handler in multiple places!